### PR TITLE
schema included properties

### PR DESCRIPTION
**Pull request by ig-robstoffers**
Adding property to let users include specific properties in the schema.

### CODE REVIEW

File: DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs [hunk 1]
```diff
@@ -606,6 +609,32 @@ private IDataSourceSchema ResolveSchema()
...
 
+            if (_schemaIncludedProperties != null)
+            {
+                List<string> propertyNames = new List<string>();
+                List<DataSourceSchemaPropertyType> propertyTypes = new List<DataSourceSchemaPropertyType>();
+
+                for (int i = 0; i < schema.PropertyNames.Length; i++)
+                {
+                    bool found = false;
+                    foreach (var includedProperty in _schemaIncludedProperties)
```
Comment (gmurray81) [inline DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs]: let's use a hash set here instead?

Comment (gmurray81) [reply DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs]: I think maybe I meant to commend on this one? Isn't this missing the null check? I find it kind of hard to read the git diffs compared to the azure devops ones.

**Review by gmurray81 — CHANGES_REQUESTED** [review]

File: DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs (hunk 1 above)
Comment (ig-robstoffers) [reply DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs]: I agree it's harder to read. I had to triple check to make sure the null check was there. But it's there a few lines above the for loop through the schema.PropertyNames.

Comment (gmurray81) [reply DataSource.DataProviders.SQLite/SQLiteDataProvider/SQLiteVirtualDataSourceDataProviderWorker.cs]: I think what is happening in some casees is I'm looking at the deltas from just one commit or something...

**Review by gmurray81 — APPROVED** [review]

**Pull request merged**